### ISSUE TITLE
TextField component's FloatingLabel should be a label element

### ISSUE THREAD

**Issue by danielstclair** [MerlinLabs/styled-material-components#190]
The internal FloatingLabel component for a TextField should be refactored to be a label component for better a11y. This would also require a refactor of the FloatingLabel CSS as labels and divs interact differently.
